packages/components/src/tooltip/index.jsx

```diff
@@ -1,65 +1,52 @@
-import React, { useCallback, useState } from 'react';
+import React, { useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import clsx from 'clsx';
-import { usePopper } from 'react-popper';
-
-import { usePopperOptions } from './hooks';
-import { POSITION_TOP } from '../feature_alert';
+import { createPopper } from '@popperjs/core';
+import { getPopperOptions } from './utils';
 import styles from './index.module.scss';
 
-const Tooltip = (props) => {
-  const [isOpen, setOpen] = useState(false);
-  const { type, text, children, className, ...cleanProps } = props;
-  const tooltipClassName = clsx(styles.tooltip, className);
-
-  const handleOpen = useCallback((event) => {
-    event.stopPropagation();
-    setOpen(true);
-  }, []);
-
-  const handleClose = useCallback((event) => {
-    event.stopPropagation();
-    setOpen(false);
-  }, []);
-
-  const [refElement, setRefElement] = useState(null);
-  const [tooltipElement, setTooltipElement] = useState(null);
-  const [arrowElement, setArrowElement] = useState(null);
-
-  const popperOptions = usePopperOptions(arrowElement, type);
-
-  const { styles: popperStyles, attributes } = usePopper(refElement, tooltipElement, popperOptions);
+const Tooltip = ({ type, text, children, className }) => {
+  const content = useRef(null);
+  const tooltip = useRef(null);
+  const instance = useRef(null);
+  const arrow = useRef(null);
+
+  const openPopper = () => {
```

Comment:
Wrap in `useCallback` if you want to keep it outside of `useEffect`, otherwise it's easy to miss `dependencies`